source/plugins/go/src/log_loss_monitoring.go

```diff
@@ -0,0 +1,303 @@
+	rotated_bytes      int64
+	existing_log_files []string
+}
+
+var FW_records map[string]FwRecord // identifier -> file names
+// var FW_bytes_logged_rotated = make(map[string]int64)   // identifier -> count
+// var FW_bytes_logged_unrotated = make(map[string]int64) // identifier -> count
+// var current_iteration_count int64 = 0
+var num_containers_on_disk int32
+
+// Shared objects
+var deleted_containers_query chan string    // There might be a lot more than 5 containers  of 10 chosen arbitrairly
+var deleted_containers_response chan string // capacity of 10 chosen arbitrairly
+
+var process_logs_mut sync.Mutex
+var read_disk_mut sync.Mutex
+var enabled bool
+
+var disabled_namespaces map[string]bool
+
+var log_loss_logger *log.Logger
+
+func init() {
+	init_log_loss_monitoring_globals()
+}
+
+// This function exists for unit tests, it lets each test reset all global variables at the start of each test
+func init_log_loss_monitoring_globals() {
+	m_bytes_logged_storage = Make_AddressableMap()
+
+	m_deletion_query_index = 0
+
+	FW_records = make(map[string]FwRecord) // identifier -> file names
+	// var FW_bytes_logged_rotated = make(map[string]int64)   // identifier -> count
+	// var FW_bytes_logged_unrotated = make(map[string]int64) // identifier -> count
+	// var current_iteration_count int64 = 0
+	num_containers_on_disk = 0
+
+	// Shared objects
+	deleted_containers_query = make(chan string, 5)    // There might be a lot more than 5 containers  of 10 chosen arbitrairly
+	deleted_containers_response = make(chan string, 5) // capacity of 10 chosen arbitrairly
+
+	process_logs_mut = sync.Mutex{}
+	read_disk_mut = sync.Mutex{}
+
+	disabled_namespaces = make(map[string]bool)
+}
+
+func SetupLogLossTracker() {
+	enabled = os.Getenv("CONTROLLER_TYPE") == "DaemonSet"
+
+	enabled = enabled && !(strings.ToLower(os.Getenv("IN_UNIT_TEST")) == "true")
+
+	// toggle env var is meant to be set by Microsoft, customer can set AZMON_DISABLE_LOG_LOSS_TRACKING through a configmap
+	enabled = enabled && (os.Getenv("AZMON_DISABLE_LOG_LOSS_TRACKING") != "true") && (os.Getenv("AZMON_DISABLE_LOG_LOSS_TRACKING_TOGGLE") != "true")
+
+	// don't count logs if stdout or stderr log collection is globally disabled
+	enabled = enabled && strings.ToLower(os.Getenv("AZMON_COLLECT_STDOUT_LOGS")) == "true" && strings.ToLower(os.Getenv("AZMON_COLLECT_STDERR_LOGS")) == "true"
+
+	if enabled {
+		log_loss_logger = createLogger("", "container-log-counts.log")
+
+		for _, excluded_namespace := range strings.Split(os.Getenv("AZMON_STDERR_EXCLUDED_NAMESPACES"), ",") {
+			disabled_namespaces[excluded_namespace] = true
+		}
+
+		for _, excluded_namespace := range strings.Split(os.Getenv("AZMON_STDOUT_EXCLUDED_NAMESPACES"), ",") {
+			disabled_namespaces[excluded_namespace] = true
+		}
+
+		write_counts_ticker := time.NewTicker(5 * time.Minute)
+		go write_telemetry(write_counts_ticker.C)
+
+		//TODO: turn this frequency down
+		track_rotations_ticker := time.NewTicker(10 * time.Second)
```

Comment:
Make duration values  5 and 10 as constants so that it can be tuned easily if required.